api/types.go

```diff
@@ -231,17 +231,18 @@ type Options struct {
 
 // Runner options which must be set when the model is loaded into memory
 type Runner struct {
-	NumCtx    int   `json:"num_ctx,omitempty"`
-	NumBatch  int   `json:"num_batch,omitempty"`
-	NumGPU    int   `json:"num_gpu,omitempty"`
-	MainGPU   int   `json:"main_gpu,omitempty"`
-	LowVRAM   bool  `json:"low_vram,omitempty"`
-	F16KV     bool  `json:"f16_kv,omitempty"`
-	LogitsAll bool  `json:"logits_all,omitempty"`
-	VocabOnly bool  `json:"vocab_only,omitempty"`
-	UseMMap   *bool `json:"use_mmap,omitempty"`
-	UseMLock  bool  `json:"use_mlock,omitempty"`
-	NumThread int   `json:"num_thread,omitempty"`
+	NumCtx     int    `json:"num_ctx,omitempty"`
+	NumBatch   int    `json:"num_batch,omitempty"`
+	NumGPU     int    `json:"num_gpu,omitempty"`
+	MainGPU    int    `json:"main_gpu,omitempty"`
+	LowVRAM    bool   `json:"low_vram,omitempty"`
+	LogitsAll  bool   `json:"logits_all,omitempty"`
+	VocabOnly  bool   `json:"vocab_only,omitempty"`
+	UseMMap    *bool  `json:"use_mmap,omitempty"`
+	UseMLock   bool   `json:"use_mlock,omitempty"`
+	NumThread  int    `json:"num_thread,omitempty"`
+	CacheTypeK string `json:"cache_type_k,omitempty"`
```

Comment:
For this PR, would it be possible to start with environment variables only? Most of our initial runtime changes are as environment variables.

Reply:
Wouldn't that mean that all your models have to be set to the same quant type?

Ideally you'd want to at least be able to configure them in the Modelfile, for example you might want certain coding or embedding models to be at f16, while conversational models all the way down at q4, likewise perhaps larger models at lower quants to be able to fit them, but smaller at higher quality to make the most of them.

Reply:
@jmorganca ?

Reply:
Sorry for the delay.

I'd rather we start with the env vars for stability for now for stability – understanding you might not want to quantize the KV cache for all models the same way. Do embedding models benefit from the KV cache anyways?